File: src/main/java/model/equipment/shield/HeavyShield.java
package model.equipment.shield;

import model.support.allEnums.DmgType;
import model.support.allEnums.EquipmentType;
import model.support.allEnums.ShieldType;

import static model.support.Constants.FIVE;
import static model.support.Constants.THREE;

public class HeavyShield extends Shield {
    public HeavyShield(double weight, int quality) {
        super(EquipmentType.SHIELD, ShieldType.HEAVY_SHIELD, weight, quality);
        price = quality / FIVE * THREE - weight * 2;
        setDefense();
    }

    private void setDefense() {
        defense.put(DmgType.BLUNT, quality * 0.20 * 3);
        defense.put(DmgType.CUT, quality * 0.20 * 3);
        defense.put(DmgType.SLASH, quality * 0.20 * 3);
        defense.put(DmgType.PIERCING, quality * 0.20 * 3);
        defense.put(DmgType.CRUSH, quality * 0.20 * 3);
    }
}

File: src/main/java/model/equipment/shield/HeavyShield.kt
package model.equipment.shield

import model.support.Constants.FIVE
import model.support.Constants.THREE
import model.support.enums.EquipmentType.SHIELD
import model.support.enums.ShieldType.HEAVY_SHIELD
import model.support.enums.Specification

class HeavyShield(
        weight: Double,
        quality: Int
) : Shield(
        SHIELD,
        HEAVY_SHIELD,
        weight,
        quality,
        quality / FIVE * THREE - weight * 2
) {
    init {
        with(defense) {
            put(Specification.BLUNT, quality * 0.20 * 3.0)
            put(Specification.CUT, quality * 0.20 * 3.0)
            put(Specification.SLASH, quality * 0.20 * 3.0)
            put(Specification.PIERCING, quality * 0.20 * 3.0)
            put(Specification.CRUSH, quality * 0.20 * 3.0)
        }
    }
}
